source/system-administrators/upgrade/upgrading-in-context-editing.rst

```diff
@@ -0,0 +1,221 @@
+
+For example, ``By <span>${contentModel.authorName_s}</span>`` is correct since the interpolated value
+is alone in its container tag, as opposed to ``<span>By ${contentModel.authorName_s}</span>``, where
+there's a piece of text inside the same element as the interpolation.
+
+Once you've located the interpolations and made sure each expression is alone in its element, replace
+the plain-html tag with a CrafterCMS macro.
+
+For example ``<span>${contentModel.authorName_s}</span>`` would become ``<@crafter.span>${contentModel.authorName_s}</@crafter.span>``.
+In addition to converting the plain tag to a macro, you must specify some additional metadata depending
+on the portion of the model that you're working with. Typically, interpolations refer to a field of
+the model, which is the missing piece of metadata on the latter example. Add the ``$field`` attribute
+to your model with the value being the field id of what you're printing.
+
+.. code-block:: html
+
+    <@crafter.span $field="authorName_s">${contentModel.authorName_s}</@crafter.span>
+
+Most HTML tags have an :ref:`equivalent macro <xbMacros>`, but if you happen to be using a *rare*
+or custom tag that doesn't have a macro, you can use the ``@crafter.tag`` macro, which will print
+any tag you specify in it's ``$tag`` argument.
+
+.. code-block:: html
+
+    <@crafter.tag $tag="author-name" $field="authorName_s">${contentModel.authorName_s}</@crafter.tag>
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Collections (components and repeat groups)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Moving on to collections, there are new powerful macros that cover the most typical cases and usages
+of component collections or repeat groups.
+
+Previous macros you're using to render a collection (e.g. ``componentContainerAttr``, ``componentContainerAttr``,
+``draggableComponent``) should be replaced with either :ref:`renderComponentCollection` or
+:ref:`renderRepeatGroup`. In case the new macros don't quite fit your use case, first,
+consider the reason why they don't and preferably update your app to work with these macros. In
+case of a valid complex scenario where these macros don't fit your use case, to enable XB you need to
+manually follow the necessary collection markup structure where you have an element representing the
+field itself, an element for each item in the collection and in the case of components, an element
+inside the item element representing the component.
+
+Like with other fields, there shouldn't be any intermediate markup in between the field elements and
+their item elements.
+
+----------
+JavaScript
+----------
+
+The approach to upgrade JavaScript applications will depend on how you first integrated your app with ICE.
+If you didn't have in-context editing on your app, head to :ref:`XB docs <xbJsApps>` to learn how to integrate.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Migrating from manual attributes
```

Comment:
Fix the case to match other similar sections. Like this: `Migrating From Manual Attributes`